Source/core/Thread.cpp

```diff
@@ -132,14 +132,14 @@ namespace Core {
 #endif
     }
 
-    ::ThreadId Thread::ThreadId()
+    thread_id Thread::ThreadId()
     {
 #ifdef __WINDOWS__
 PUSH_WARNING(DISABLE_WARNING_CONVERSION_TO_GREATERSIZE)
```

Comment:
thread_id is DWORD and GetCurrentThreadId() return DWORD, why would there be a warning?